app/models/project_step.rb

```diff
@@ -52,6 +52,9 @@ class ProjectStep < ActiveRecord::Base
 
   validates :project_id, presence: true
 
+  before_save :before_save  #handles original date logic
```

Comment:
Please rename this callback method to be more descriptive. e.g. `handle_original_date_logic`